src/panels/config/automation/ha-automation-sidebar.ts

```diff
@@ -0,0 +1,401 @@
+              .localize=${this.hass.localize}
+              .warnings=${this._warnings}
+            >
+            </ha-automation-editor-warning>`
+          : nothing}
+        ${this.config.type === "trigger"
+          ? html`<ha-automation-trigger-editor
+              class="sidebar-editor"
+              .hass=${this.hass}
+              .trigger=${this.config.config as Trigger}
+              @value-changed=${this._valueChangedSidebar}
+              .uiSupported=${this.config.uiSupported}
+              .showId=${this._requestShowId}
+              .yamlMode=${this._yamlMode}
+              .disabled=${this.disabled}
+              @ui-mode-not-available=${this._handleUiModeNotAvailable}
+            ></ha-automation-trigger-editor>`
+          : this.config.type === "condition" &&
+              (this._yamlMode || !CONDITION_BUILDING_BLOCKS.includes(type))
+            ? html`
+                <ha-automation-condition-editor
+                  class="sidebar-editor"
+                  .hass=${this.hass}
+                  .condition=${this.config.config as Condition}
+                  .yamlMode=${this._yamlMode}
+                  .uiSupported=${this.config.uiSupported}
+                  @value-changed=${this._valueChangedSidebar}
+                  .disabled=${this.disabled}
+                  @ui-mode-not-available=${this._handleUiModeNotAvailable}
+                ></ha-automation-condition-editor>
+              `
+            : this.config.type === "action" &&
+                (this._yamlMode || !ACTION_BUILDING_BLOCKS.includes(type))
+              ? html`
+                  <ha-automation-action-editor
+                    class="sidebar-editor"
+                    .hass=${this.hass}
+                    .action=${this.config.config as Action}
+                    .yamlMode=${this._yamlMode}
+                    .uiSupported=${this.config.uiSupported}
+                    @value-changed=${this._valueChangedSidebar}
+                    sidebar
+                    narrow
+                    .disabled=${this.disabled}
+                    @ui-mode-not-available=${this._handleUiModeNotAvailable}
+                  ></ha-automation-action-editor>
+                `
+              : description
+                ? html`<div class="card-content">${description}</div>`
+                : nothing}
+      </ha-card>
+    `;
+  }
+
+  private _handleUiModeNotAvailable(ev: CustomEvent) {
+    this._warnings = handleStructError(this.hass, ev.detail).warnings;
+    if (!this._yamlMode) {
+      this._yamlMode = true;
+    }
+  }
+
+  private _valueChangedSidebar(ev: CustomEvent) {
+    ev.stopPropagation();
+
+    this.config?.save(ev.detail.value);
+
+    if (this.config) {
+      fireEvent(this, "value-changed", {
+        value: {
+          ...this.config,
+          config: ev.detail.value,
+        },
+      });
+    }
+  }
+
+  private _closeSidebar() {
+    this.config?.close();
+    this.config = undefined;
```

Comment:
Shouldn't the parent set this to undefined instead of the child? Will the parent not set this to his stored value again now?

Reply:
Yes I think we can remove this.